test/jdk/javax/imageio/plugins/jpeg/WriteJPEGThumbnailTest.java

```diff
@@ -0,0 +1,158 @@
+
+    public WriteJPEGThumbnailTest(int thumbWidth, int thumbHeight) {
+        this.thumbWidth = thumbWidth;
+        this.thumbHeight = thumbHeight;
+    }
+
+    public boolean run() throws Exception {
+        System.out.println("Testing thumbnail " + thumbWidth + "x" + thumbHeight + "...");
+        try {
+            byte[] jpegData;
+            BufferedImage thumbnail;
+            try (ByteArrayOutputStream byteOut = new ByteArrayOutputStream()) {
+                try {
+                    thumbnail = writeImage(byteOut);
+                } finally {
+                    jpegData = byteOut.toByteArray();
+                }
+            }
+
+            ImageReader reader = getJPEGImageReader();
+            ImageInputStream stream = ImageIO.createImageInputStream(new ByteArrayInputStream(jpegData));
+            reader.setInput(stream);
+            assertEquals(1, reader.getNumThumbnails(0));
+
+            // we may have a subset of our original thumbnail, that's OK
+            BufferedImage readThumbnail = reader.readThumbnail(0, 0);
+            for (int y = 0; y < readThumbnail.getHeight(); y++) {
+                for (int x = 0; x < readThumbnail.getWidth(); x++) {
+                    int rgb1 = thumbnail.getRGB(x, y);
+                    int rgb2 = readThumbnail.getRGB(x, y);
+                    assertEquals(rgb1, rgb2);
+                }
+            }
+            System.out.println("\tTest passed");
+        } catch(Throwable e) {
+            e.printStackTrace();
+            return false;
+        }
+        return true;
+    }
+
+    private BufferedImage writeImage(OutputStream out) throws IOException {
+        BufferedImage thumbnail = createImage(thumbWidth, thumbHeight);
+        BufferedImage bi = createImage(thumbnail.getWidth() * 10, thumbnail.getHeight() * 10);
+
+        ImageWriter writer = ImageIO.getImageWritersByFormatName("jpeg").next();
+
+        try (ImageOutputStream outputStream = ImageIO.createImageOutputStream(out)) {
+            writer.setOutput(outputStream);
+
+            // Write the main image
+            IIOImage img = new javax.imageio.IIOImage(bi, List.of(thumbnail), null);
+            writer.write(null, img, null);
+        } finally {
+            writer.dispose();
+        }
+        return thumbnail;
+    }
+
+    private static BufferedImage createImage(int width, int height) {
+        BufferedImage bi = new BufferedImage(width, height, BufferedImage.TYPE_INT_RGB);
+        Graphics2D g = bi.createGraphics();
+        double sx = ((double)width) / 1000.0;
+        double sy = ((double)height) / 1000.0;
+        g.transform(AffineTransform.getScaleInstance(sx, sy));
+        g.setColor(Color.red);
+        g.fillRect(0,0,100,100);
+        g.setColor(Color.green);
+        g.fillRect(900,0,900,100);
+        g.setColor(Color.orange);
+        g.fillRect(0,900,100,100);
+        g.setColor(Color.magenta);
+        g.fillRect(900,900,100,100);
```

Comment:
Spaces after commas, upper-case color constants:
```suggestion
        g.setColor(Color.RED);
        g.fillRect(0, 0, 100, 100);
        g.setColor(Color.GREEN);
        g.fillRect(900, 0, 900, 100);
        g.setColor(Color.ORANGE);
        g.fillRect(0, 900, 100, 100);
        g.setColor(Color.MAGENTA);
        g.fillRect(900, 900, 100, 100);
```